### PR TITLE
Adjust action groups to moved modules

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
The docker modules, kubevirt modules and the k8s module are moving to other collections:

1. The docker modules are moving from community.general to [community.docker](https://github.com/ansible-collections/community.docker/) (already in progress, new collection already released).
2. The kubevirt modules are moving from community.general to [community.kubevirt](https://github.com/ansible-collections/community.kubevirt) (in preparation, collection not yet published).
3. ~~The k8s module is moving from community.kubernetes to [community.okd](https://github.com/ansible-collections/community.okd) (already in progress, collection with module already released).~~
4. The community.k8s collection will be renamed to [kubernetes.core](https://github.com/ansible-collections/kubernetes.core) (already in progress, new collection already released).

Right now, support for the docker and k8s action groups / module default groups does need to know in which collection the affected modules / action plugins are contained. Therefore, lib/ansible/executor/module_common.py needs to be updated (and this change backported to stable-2.10) for the moves to work flawlessly, and to avoid breaking playbooks and roles.

The old names must not be removed either to avoid breakage with older (current) versions of the collections.

CC @s-hertel @gundalow @geerlingguy @tima

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/executor/module_common.py

### CODE REVIEW

**Comment by felixfontein** [comment]
@geerlingguy @tima similar PRs like ansible-collections/community.docker#17 will be needed in your collections as well for this to work.

**Comment by felixfontein** [comment]
@gundalow I think the tests are broken (maybe just temporarily), unrelated to this PR.

**Comment by geerlingguy** [comment]
@felixfontein the k8s module is not moving to the okd collection; that collection just has an alias/override of the k8s module customized for OpenShift.

**Comment by felixfontein** [comment]
@geerlingguy thanks! I've removed community.okd from this PR then.

**Comment by geerlingguy** [comment]
@felixfontein though the oc plugin is moving from community.general to community.okd, and the k8s_auth plugin moved from community.kubernetes to community.okd.

**Comment by felixfontein** [comment]
@geerlingguy ok, in that case I reverted the last commit :) It's then needed for the k8s_auth module. (Module defaults are not used for non-action plugins, so the oc connection plugin doesn't matter.)

File: lib/ansible/executor/module_common.py [hunk 1]
```diff
@@ -1357,9 +1357,9 @@ def get_action_args_with_defaults(action, args, defaults, templar, redirected_na
         'aws': ['amazon.aws', 'community.aws'],
         'azure': ['azure.azcollection'],
         'cpm': ['wti.remote'],
-        'docker': ['community.general'],
+        'docker': ['community.general', 'community.docker'],
```
Comment (bcoca) [inline lib/ansible/executor/module_common.py:1360]: not a blocker to this PR, but this should be moved to config (really should not be a thing in core at all, but setlingling for moving into config).

Comment (felixfontein) [reply lib/ansible/executor/module_common.py:1360]: I was assuming that this is a placeholder anyway until the action groups feature is fully designed and implemented.

Comment (sivel) [reply lib/ansible/executor/module_common.py:1360]: I believe that is tentatively slated for 2.11

**Review by gundalow — APPROVED** [review]
Looks OK to me

**Comment by felixfontein** [comment]
@geerlingguy @bcoca @sivel @gundalow thanks for reviewing!

**Pull request merged**